src/chunkserver/op_request.h

```diff
@@ -357,6 +360,36 @@ class PasteChunkInternalRequest : public ChunkOpRequest {
     butil::IOBuf data_;
 };
 
+class ScanChunkRequest : public ChunkOpRequest {
+ public:
+    ScanChunkRequest(uint64_t index, PeerId peer) :
+       ChunkOpRequest(), index_(index), peer_(peer) {}
+    ScanChunkRequest(std::shared_ptr<CopysetNode> nodePtr,
+                     ScanManager* scanManager,
+                     const ChunkRequest *request,
+                     ChunkResponse *response,
+                     ::google::protobuf::Closure *done) :
+        ChunkOpRequest(nodePtr,
+                      NULL,
+                      request,
+                      response,
+                      done),
+                      scanManager_(scanManager) {}
+    virtual ~ScanChunkRequest() = default;
+
+    void OnApply(uint64_t index, ::google::protobuf::Closure *done) override;
+    void OnApplyFromLog(std::shared_ptr<CSDataStore> datastore,
+                        const ChunkRequest &request,
+                        const butil::IOBuf &data) override;
+ private:
+    void BuildRepScanMap(LogicPoolID pollId, ChunkID chunkId, uint64_t index, 
+                         uint64_t offset, uint64_t len, char* readBuf);
+    void SendScanMapToLeader();
+    ScanManager* scanManager_;     
+    uint64_t index_;
```

Comment:
what's the meaning of `index_`, raft apply index ?

Reply:
yes